File: backend/src/main/java/com/NBE3_4_2_Team4/domain/board/search/entity/NewsSearchResult.java
package com.NBE3_4_2_Team4.domain.board.search.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Embeddable
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class NewsSearchResult {
    private String title;
    private String link;
    private String description;
}

File: backend/src/main/java/com/NBE3_4_2_Team4/domain/board/search/entity/NewsSearchResult.kt
package com.NBE3_4_2_Team4.domain.board.search.entity

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import jakarta.persistence.Embeddable

@Embeddable
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
data class NewsSearchResult(
    val title: String = "",
    val link: String = "",
    val description: String = ""
)
